src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -512,29 +511,29 @@ public ResponseEntity<StudentExam> getStudentExamForSummary(@PathVariable Long c
     }
 
     /**
-     * GET /courses/{courseId}/exams/{examId}/student-exams/grade-summary : Return student exam result, aggregate points, assessment result
+     * GET /courses/{courseId}/exams/{examId}/student-exams/{studentExamId}/grade-summary : Return student exam result, aggregate points, assessment result
      * for a student exam and grade calculations if the exam is assessed. Only instructors can use userId parameter to get exam results of other users,
      * if the caller is a student, userId should either be the user id of the caller or empty.
      * <p>
      * Does not return the student exam itself to save bandwidth.
      * <p>
      * See {@link StudentExamWithGradeDTO} for more explanation.
      *
-     * @param courseId  the course to which the student exam belongs to
-     * @param examId    the exam to which the student exam belongs to
-     * @param userId    the user id of the student whose grade summary is requested
-     * @param isTestRun whether the student exam belongs to a test run by an instructor or not
+     * @param courseId      the course to which the student exam belongs to
+     * @param examId        the exam to which the student exam belongs to
+     * @param studentExamId the id of the student exam whose grade summary is requested
+     * @param userId        the user id of the student whose grade summary is requested
      * @return the ResponseEntity with status 200 (OK) and with the StudentExamWithGradeDTO instance without the student exam as body
      */
-    @GetMapping("courses/{courseId}/exams/{examId}/student-exams/grade-summary")
+    @GetMapping("courses/{courseId}/exams/{examId}/student-exams/{studentExamId}/grade-summary")
     @EnforceAtLeastStudent
-    public ResponseEntity<StudentExamWithGradeDTO> getStudentExamGradesForSummary(@PathVariable Long courseId, @PathVariable Long examId,
-            @RequestParam(required = false) Long userId, @RequestParam(required = false, defaultValue = "false") boolean isTestRun) {
+    public ResponseEntity<StudentExamWithGradeDTO> getStudentExamGradesForSummary(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable Long studentExamId,
```

Comment:
```suggestion
    public ResponseEntity<StudentExamWithGradeDTO> getStudentExamGradesForSummary(@PathVariable long courseId, @PathVariable long examId, @PathVariable long studentExamId,
```
Since they are required, we can use the base type